### PR TITLE
Update typings for `diff-match-patch`

### PR DESCRIPTION
**Pull request by thasner**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: diff-match-patch/diff-match-patch.d.ts [hunk 1]
```diff
@@ -31,6 +36,8 @@ declare module "diff-match-patch" {
         diff_levenshtein(diffs: Diff[]): number;
         diff_toDelta(diffs: Diff[]): string;
         diff_fromDelta(text1: string, delta: string): Diff[];
+        diff_linesToChars_(text1: string, text2: string): Encoded;
```
Comment (thasner) [inline diff-match-patch/diff-match-patch.d.ts]: Function prototype: https://code.google.com/p/google-diff-match-patch/source/browse/trunk/javascript/diff_match_patch_uncompressed.js#443

Return type: https://code.google.com/p/google-diff-match-patch/source/browse/trunk/javascript/diff_match_patch_uncompressed.js#490

File: diff-match-patch/diff-match-patch.d.ts [hunk 2]
```diff
@@ -31,6 +36,8 @@ declare module "diff-match-patch" {
...
+        diff_charsToLines_(diffs: Diff[], lineArray: string[]);
```
Comment (thasner) [inline diff-match-patch/diff-match-patch.d.ts]: Function prototype: https://code.google.com/p/google-diff-match-patch/source/browse/trunk/javascript/diff_match_patch_uncompressed.js#501

**Comment by vsiao** [comment]
This API is private. I don't think it should go in the typings.

**Comment by mhegazy** [comment]
Private APIs should not be added to the declaration file. Please file an issue for the package author to mark/document it as public.

**Comment by bavardage** [comment]
it's referenced in https://code.google.com/p/google-diff-match-patch/wiki/LineOrWordDiffs as the recommended way to do line-level diffing.

**Comment by bavardage** [comment]
https://code.google.com/p/google-diff-match-patch/issues/detail?id=111&thanks=111&ts=1466456922 fysa

**Pull request closed without merging**